src/main/java/Duke.java

```diff
@@ -1,10 +1,238 @@
+            areValidParameters(new String[] {parameters});
+
+            Item newTodo = new Remind(parameters);
+            items.add(newTodo);
+
+            printAddItemMessage(newTodo);
+        } catch (Exception err) {
+            throw new DukeException(Message.ERROR_REMIND_MISSING_PARAMETER.toString());
+        }
+    }
+
+    /**
+     * Adds a Deadline item into the list.
+     * 
+     * @param parameters Gets the description and deadline of the item.
+     * @throws DukeException when there are missing parameters or areValidParameters check fails
+     */
+    private static void addDeadline(String parameters) throws Exception {
+        try {
+            String[] attributes = parameters.split(" /by ", 2);
+            String description = attributes[0];
+            String by = attributes[1];
+
+            areValidParameters(new String[] {description, by});
+
+            Item newDeadline = new Deadline(description, by);
+            items.add(newDeadline);
+
+            printAddItemMessage(newDeadline);
+        } catch (Exception err) {
+            throw new DukeException(Message.ERROR_DEADLINE_MISSING_PARAMETER.toString());
+        }
+    }
+
+    /**
+     * Adds a Event item into the list.
+     * 
+     * @param parameters Gets the description and from/to of the item.
+     * @throws DukeException when there are missing parameters or areValidParameters check fails
+     */
+    private static void addEvent(String parameters) throws Exception {
+        try {
+            String[] attributes = parameters.split(" /from ", 2);
+            String description = attributes[0];
+            
+            attributes = attributes[1].split(" /to ", 2);
+            String from = attributes[0];
+            String to = attributes[1];
+
+            areValidParameters(new String[] {description, from, to});
+
+            Item newEvent = new Event(description, from, to);
+            items.add(newEvent);
+
+            printAddItemMessage(newEvent);
+        } catch (Exception err) {
+            throw new DukeException(Message.ERROR_EVENT_MISSING_PARAMETER.toString());
+        }
+    }
+
+    /**
+     * Prints the welcome message when application is launched.
+     */
+    private static void printWelcomeMessage() {
+        System.out.println(Message.LINE);
+        System.out.println(Message.INFO_WELCOME);
+        System.out.println(Message.LINE);
+    }
+
+    /**
+     * Prints the message after an item is added.
+     * 
+     * @param item Prints the description and status of the item
+     */
+    private static void printAddItemMessage(Item item) {
+        System.out.println(Message.INFO_ITEM_ADD);
+        System.out.println(item);
+        System.out.printf(Message.INFO_ITEM_COUNT.toString(), items.size());
+    }
+
+    /**
+     * Prints a bye message and exits the application.
+     */
+    private static void exitProgram() {
+        System.out.println(Message.INFO_EXIT);
+        System.exit(0);
     }
```

Comment:
Good job following the method & variable naming conventions and Javadoc comments structure!